src/main/java/com/epam/test_generator/services/JiraService.java

```diff
@@ -0,0 +1,332 @@
+
+    @Autowired
+    private CaseDAO caseDAO;
+
+    @Autowired
+    private UserDAO userDAO;
+
+    @Autowired
+    private RemovedIssueDAO removedIssueDAO;
+
+    private static final Integer FIRST = 0;
+
+
+    /**
+     * Creates project from Jira in the system with specified jira stories (suits in BDD) and all
+     * subtasks (cases in BDD) of these stories.
+     *
+     * @param stories collection of Jira stories
+     */
+    public void createProjectWithAttachments(List<JiraStory> stories) throws JiraException {
+        if (!stories.isEmpty()) {
+
+            String projectKey = stories.get(FIRST).getJiraProjectKey();
+            JiraProject projectByJiraKey = jiraProjectDAO.getProjectByJiraKey(projectKey);
+            createProjectFromJiraProject(projectByJiraKey);
+            addStoriesToExistedProject(stories);
+        }
+    }
+
+    /**
+     * Creates specified jira stories (suits in BDD) and all subtasks (cases in BDD) of these stories
+     * for already existed in BDD project.
+     *
+     * @param stories collection of Jira stories
+     */
+    public void addStoriesToExistedProject(List<JiraStory> stories) throws JiraException {
+        stories.forEach(this::createSuitFromJiraStory);
+        String query = String.format(
+                "project=%s AND status in (Open, \"In Progress\", Reopened, Verified) AND type=sub-task",
+                stories.get(FIRST).getJiraProjectKey());
+
+        List<JiraSubTask> jiraSubTasks = jiraSubStoryDAO.getJiraSubtoriesByFilter(query);
+
+        jiraSubTasks.stream()
+                .filter(subTask -> suitDAO.findByJiraKey(subTask.getJiraParentKey()) != null)
+                .filter(subTask -> caseDAO.findByJiraKey(subTask.getJiraKey()) == null)
+                .forEach(this::createCaseFromJiraSubTask);
+    }
+
+
+    public List<JiraStory> getStories(String jiraProjectKey) throws JiraException {
+        return jiraStoryDAO.getStories(jiraProjectKey);
+    }
+
+
+    /**
+     * Returns stories which are not in the system from jira from specified project
+     * that is already in the system.
+     *
+     * @param projectId id project in BDD
+     * @return collection of Jira stories
+     */
+    public List<JiraStory> getJiraStoriesFromJiraProjectByProjectId(Long projectId) throws JiraException {
+        String projectKey = projectDAO.getOne(projectId).getJiraKey();
+        return jiraStoryDAO.getNonexistentStoriesByProject(projectKey);
+    }
+
+    /**
+     * Creates and associates a new project in BDD to Jira project.
+     *
+     * @param jiraProject - new project from Jira
+     */
+    private void createProjectFromJiraProject(JiraProject jiraProject) {
+        Project project = new Project();
+        project.setName(jiraProject.getName());
+        project.setDescription(jiraProject.getDescription());
+        project.setJiraKey(jiraProject.getJiraKey());
+        project.setActive(true);
+        User user = userDAO.findByEmail("admin@mail.com");
```

Comment:
Does it correct that you have constant for user email? If the answer yes, I think it can be extract as a constant variable.

Reply:
Done